libs/s25main/buildings/nobMilitary.cpp

```diff
@@ -149,6 +168,15 @@ nobMilitary::nobMilitary(SerializedGameData& sgd, const unsigned obj_id)
     sgd.PopObjectContainer(troops, GO_Type::NofPassivesoldier);
     sgd.PopObjectContainer(far_away_capturers, GO_Type::NofAttacker);
 
+    if(sgd.GetGameDataVersion() < 10)
+    {
+        desired_troops.fill(GetMaxTroopsCt());
+    } else
+    {
+        for(unsigned i = 0; i < NUM_SOLDIER_RANKS; ++i)
+            desired_troops[i] = sgd.PopUnsignedInt();
```

Comment:
```suggestion
        sgd.PopContainer(desired_troops);
```

Reply:
This function doesn't work for fixed size containers, used the one from `helpers` again.